### PR TITLE
`lhs_constants` option for #1359

### PR DESCRIPTION
**Pull request by iccir**
I implemented a new option called `lhs_constants` which defaults to `true`. I thought about having `comparisons: false` turn off left-hand side constants, but didn't like this as non-comparison operators (`*`, `&`, `|`, `^`) are also affected.

That said, several `comparisons` optimizations assume `lhs_constants: true`. I tried to decouple the two, but I could use some guidance: should we add more complexity in `compress/index.js` to allow `comparisons: true, lhs_constants: false`? My assumption is that most people will either want both `true` or both `false`.

I audited all failing test cases. I added `lhs_constants: true` for those which tested comparisons (or those that already had several other options set to `true`). When a test case focused on non-comparison optimizations, I reversed the order in the expected result.

### CODE REVIEW

File: lib/compress/index.js [hunk 1]
```diff
@@ -2174,6 +2178,19 @@ def_optimize(AST_Binary, function(self, compressor) {
                 self.left = make_node(AST_Undefined, self.left).optimize(compressor);
                 if (self.operator.length == 2) self.operator += "=";
             }
+        } else if (compressor.option("typeofs")
```
Comment (iccir) [inline lib/compress/index.js:2181]: This block of code is only applicable for `comparisons: true, typeofs: true, lhs_constants: false`. I'm not sure if the added complexity is worth it.

File: test/compress/asm.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ asm_mixed: {
         join_vars: true,
         keep_fargs: true,
         keep_fnames: false,
+        lhs_constants: true,
```
Comment (iccir) [inline test/compress/asm.js:15]: When a test case already had several `true` options, I didn't mind adding `lhs_constants: true`.

File: test/compress/class-properties.js [hunk 3]
```diff
@@ -182,11 +182,11 @@ static_means_execution: {
         let x = 0;
         // Does not get inlined as it contains an immediate side effect
         class WithStaticProps {
-            static prop = (x = 0 === x ? 1 : "FAIL")
```
Comment (iccir) [inline test/compress/class-properties.js:185]: In this case, I opted to move the `0` back to the right-hand side. This test case is testing code unrelated to comparisons.

**Comment by fabiosantoscode** [comment]
Nicely done, thanks! I'll release during the week

**Pull request merged**